Sqlite

jkost opened the pull request.

minborg commented:
@jkost Would it be possible for you to create a pull request against the 'develop' branch? If not, I can manually pull in your PR to the develop branch but then we will lose the information that you've created them.

minborg commented:
Some of the classes could be manually merged to create Sqllite support

jkost commented:
Hallo Per-Ake,

sorry for not contributing much the last year. I was busy with other
things. I have never managed to make it work.

I 'll try in the evening or tomorrow to download the latest version,
because with previous versions I got stuck.

If I manage to make it work, then I 'll continue with another schema-less
DB, HSQLDB.

Thanks.

John.

On 31 October 2017 at 04:56, Per-Åke Minborg <notifications@github.com>
wrote:

> Some of the classes could be manually merged to create Sqllite support
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/speedment/speedment/pull/185#issuecomment-340653041>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAaaYLA8Cyi_Lc1zBa3M0mDv5fsYO_ceks5sxpp9gaJpZM4ILA2S>
> .
>

jkost commented:
Hallo Per-Ake,

it's been sometime I haven't worked on the project. Version 3.0.0 returns:

git branch -a
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/develop-2.3
  remotes/origin/develop-3.0
  remotes/origin/develop-3.1
  remotes/origin/feature-advanced-ui
  remotes/origin/feature-crud-streams
  remotes/origin/feature-java9
  remotes/origin/feature-jigsaw
  remotes/origin/feature-nosql
  remotes/origin/feature-plugrender
  remotes/origin/feature-transaction2
  remotes/origin/feature-transaction3
  remotes/origin/feature-transactions
  remotes/origin/master

In version 2.3.0 which I had done my changes in, I had created my own
branch sqlite:

git branch -a
  dev
  master
* sqlite
  remotes/origin/HEAD -> origin/master
  remotes/origin/develop
  remotes/origin/develop-advanced-gui
  remotes/origin/develop-crud-streams
  remotes/origin/develop-modules
  remotes/origin/develop-nosql
  remotes/origin/develop-transactions
  remotes/origin/master

and I had done a number of changes:

    modified:   pom.xml
    modified:   src/main/java/com/speedment/db/DbmsHandler.java
    modified:
src/main/java/com/speedment/internal/core/config/dbms/StandardDbmsType.java
    modified:
src/main/java/com/speedment/internal/core/db/AbstractRelationalDbmsHandler.java
    modified:
src/main/java/com/speedment/internal/ui/controller/SceneController.java
New classes:

src/main/java/com/speedment/config/db/mapper/identity/NumericIdentityMapper.java

src/main/java/com/speedment/internal/core/config/dbms/SqliteDbmsType.java
    src/main/java/com/speedment/internal/core/db/SqliteDbmsHandler.java

src/main/java/com/speedment/internal/core/manager/sql/SqliteSpeedmentPredicateView.java

So, when you say to develop a PR to the 'develop' branch, which 'develop'
branch exactly?

  remotes/origin/develop-2.3
  remotes/origin/develop-3.0
  remotes/origin/develop-3.1

I 'd like to keep sqlite branch (and hsqldb branch), but I 'll clone from
develop-2.3 and try develop-3.0, too, to see if it makes my life easier.

Hope I didn't misunderstand you.

Thanks.

John.

On 31 October 2017 at 11:18, John Kostaras <jkostaras@gmail.com> wrote:

> Hallo Per-Ake,
>
> sorry for not contributing much the last year. I was busy with other
> things. I have never managed to make it work.
>
> I 'll try in the evening or tomorrow to download the latest version,
> because with previous versions I got stuck.
>
> If I manage to make it work, then I 'll continue with another schema-less
> DB, HSQLDB.
>
> Thanks.
>
> John.
>
> On 31 October 2017 at 04:56, Per-Åke Minborg <notifications@github.com>
> wrote:
>
>> Some of the classes could be manually merged to create Sqllite support
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/speedment/speedment/pull/185#issuecomment-340653041>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAaaYLA8Cyi_Lc1zBa3M0mDv5fsYO_ceks5sxpp9gaJpZM4ILA2S>
>> .
>>
>
>

minborg commented:
@jkost Great news. Welcome back again! So if I could update you a bit, there is a new module in Speedment called "connector-parent" which now contains only an H2 connector (which is not working but work in progress). See https://github.com/speedment/speedment/tree/develop-3.0/connector-parent

Please create new module(s) for the database type(s) you are contributing with. So you could take the H2 module and copy it and work with it from there.

It is best if you just start from the branch develop-3.0 and copy in any work you have previously done. Since 2.x, there were many changes. Also, please create you PRs against the develop-3.0 branch.

Thanks! /Per

jkost commented:
Hallo. I 've spent some time trying to figure out why I get this error:

`Execution default-cli of goal com.speedment:speedment-maven-plugin:3.0.19-SNAPSHOT:tool failed: The database type SQLite is not registered with the DbmsHandlerComponent`

I don't know what is missing. Any help/hints is/are appreciated.

`On branch develop-3.0
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   archetype-parent/speedment-archetype-sqlite/src/main/resources/META-INF/maven/archetype.xml
	new file:   archetype-parent/speedment-archetype-sqlite/src/main/resources/archetype-resources/pom.xml
	new file:   archetype-parent/speedment-archetype-sqlite/src/main/resources/archetype-resources/src/main/java/Main.java
	new file:   connector-parent/sqlite/pom.xml
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/SqliteBundle.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/SqliteComponent.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteComponentImpl.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteDbmsMetadataHandler.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteDbmsOperationHandler.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteDbmsType.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteSpeedmentPredicateView.java
	new file:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/package-info.java
	new file:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/sqlite/SqliteDbmsMetadataHandler.java
	new file:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/sqlite/SqliteDbmsOperationHandler.java
	new file:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/sqlite/SqliteDbmsType.java

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   archetype-parent/README.md
	modified:   archetype-parent/pom.xml
	modified:   archetype-parent/speedment-archetype-sqlite/src/main/resources/META-INF/maven/archetype.xml
	modified:   archetype-parent/speedment-archetype-sqlite/src/main/resources/archetype-resources/pom.xml
	modified:   build-parent/maven-plugin/pom.xml
	modified:   connector-parent/h2/src/main/java/com/speedment/connector/h2/internal/H2ComponentImpl.java
	modified:   connector-parent/pom.xml
	modified:   connector-parent/sqlite/pom.xml
	modified:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/SqliteComponent.java
	modified:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteComponentImpl.java
	modified:   connector-parent/sqlite/src/main/java/com/speedment/connector/sqlite/internal/SqliteDbmsType.java
	modified:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/StandardDbmsTypes.java
	modified:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/sqlite/SqliteDbmsMetadataHandler.java
	modified:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/sqlite/SqliteDbmsOperationHandler.java
	modified:   runtime-parent/runtime-core/src/main/java/com/speedment/runtime/core/internal/db/sqlite/SqliteDbmsType.java

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	archetype-parent/speedment-archetype-sqlite/pom.xml
	tool-parent/tool-core/src/main/json/
`

minborg commented:
Hi @jkost !
Great to see you again. I think the best thing would be to manually copy your files into the latest version of the branch of develop-3.0. New database support files now goes into a separate module named connector-parent as I can see that you have understood. Great!

If you want your new DbmsType to work, you need to install it with the `DbmsHandlerComponent`. You can see how this is done in the class `RuntimeBundle`
